src/backend/gl/src/command.rs

```diff
@@ -326,44 +327,44 @@ impl RawCommandBuffer {
 
     fn update_blend_targets(&mut self, blend_targets: &Vec<pso::ColorBlendDesc>) {
         let max_blend_slots = blend_targets.len();
+        if max_blend_slots == 0 {
+            return;
+        }
 
-        if max_blend_slots > 0 {
-            match self.cache.blend_targets {
-                Some(ref mut cached) => {
-                    if cached.len() < max_blend_slots {
-                        cached.resize(max_blend_slots, None);
-                    }
-                }
-                None => {
-                    self.cache.blend_targets = Some(vec![None; max_blend_slots]);
-                }
-            };
+        if self.cache.blend_targets.len() < max_blend_slots {
+            self.cache.blend_targets.resize(max_blend_slots, None);
         }
 
-        for (slot, blend_target) in blend_targets.iter().enumerate() {
-            let mut update_blend = false;
-            if let Some(ref mut cached_targets) = self.cache.blend_targets {
-                if let Some(cached_target) = cached_targets.get(slot) {
-                    match cached_target {
-                        &Some(ref cache) => {
-                            if cache != blend_target {
-                                update_blend = true;
-                            }
-                        }
-                        &None => {
-                            update_blend = true;
-                        }
+        let mut all_targets_same = true;
+        for blend_target in &blend_targets[1..] {
+            if blend_target != &blend_targets[0] {
+                all_targets_same = false;
+                break;
+            }
+        }
+
+        if all_targets_same {
+            for cached_target in &mut self.cache.blend_targets {
+                *cached_target = Some(blend_targets[0]);
+            }
+            self.push_cmd(Command::BindAllBlendSlots(blend_targets[0]));
```

Comment:
I guess this is wrong because it shouldn't set the blend state again if all of the cached targets also match... really the cache state should be in one of two states (all set to X settings, per-targets set to X settings), but that would require potentially dropping and re-creating a Vec or getting pretty messy.  I think I can at least avoid this command if all of the blend slots match already.